packages/eddsa-pcd/src/EdDSAPCD.ts

```diff
@@ -0,0 +1,146 @@
+import { v4 as uuid } from "uuid";
+import { EdDSACardBody } from "./CardBody";
+
+export const EdDSAPCDTypeName = "eddsa-pcd";
+
+export interface EdDSAPCDArgs {
+  privateKey: StringArgument;
+  message: StringArrayArgument;
+  id: StringArgument;
+}
+
+export interface EdDSAPCDClaim {
+  publicKey: [string, string];
+  message: bigint[];
+}
+
+export interface EdDSAPCDProof {
+  signature: string;
+}
+
+export class EdDSAPCD implements PCD<EdDSAPCDClaim, EdDSAPCDProof> {
+  type = EdDSAPCDTypeName;
+  claim: EdDSAPCDClaim;
+  proof: EdDSAPCDProof;
+  id: string;
+
+  public constructor(id: string, claim: EdDSAPCDClaim, proof: EdDSAPCDProof) {
+    this.id = id;
+    this.claim = claim;
+    this.proof = proof;
+  }
+}
+
+const fromHexString = (hexString: string) => Buffer.from(hexString, "hex");
+
+const toHexString = (bytes: Uint8Array) => Buffer.from(bytes).toString("hex");
+
+export async function prove(args: EdDSAPCDArgs): Promise<EdDSAPCD> {
+  if (!args.privateKey.value) {
+    throw new Error("No private key value provided");
+  }
+
+  if (!args.message.value) {
+    throw new Error("No message value provided");
+  }
+
+  let message;
+  try {
+    message = args.message.value.map((fieldStr: string) => BigInt(fieldStr));
+  } catch (e) {
+    throw new Error("Could not convert message contents to bigint type");
+  }
+
+  const id = typeof args.id.value === "string" ? args.id.value : uuid();
+
+  const prvKey = fromHexString(args.privateKey.value);
+
+  const eddsa = await buildEddsa();
+  const poseidon = await buildPoseidon();
+
+  const hashedMessage = poseidon(message);
+  const publicKey: [string, string] = eddsa
+    .prv2pub(prvKey)
+    .map(toHexString) as [string, string];
+  const signature = toHexString(
+    eddsa.packSignature(eddsa.signPoseidon(prvKey, hashedMessage))
+  );
+
+  return new EdDSAPCD(id, { message, publicKey }, { signature });
+}
+
+export async function verify(pcd: EdDSAPCD): Promise<boolean> {
+  const eddsa = await buildEddsa();
+  const poseidon = await buildPoseidon();
+
+  const signature = eddsa.unpackSignature(fromHexString(pcd.proof.signature));
+  const pubKey = pcd.claim.publicKey.map(fromHexString);
+  const hashedMessage = poseidon(pcd.claim.message);
+
+  return eddsa.verifyPoseidon(hashedMessage, signature, pubKey);
+}
+
+function replacer(key: any, value: any): any {
+  if (key === "message") {
+    return value.map((num: bigint) => num.toString(16));
+  } else {
+    return value;
+  }
+}
+
+function reviver(key: any, value: any): any {
+  if (key === "message") {
+    return value.map((str: string) => BigInt(`0x${str}`));
+  } else {
+    return value;
+  }
+}
+
+export async function serialize(
+  pcd: EdDSAPCD
+): Promise<SerializedPCD<EdDSAPCD>> {
+  return {
+    type: EdDSAPCDTypeName,
+    pcd: JSON.stringify(pcd, replacer)
+  };
+}
+
+export async function deserialize(serialized: string): Promise<EdDSAPCD> {
+  return JSON.parse(serialized, reviver);
+}
```

Comment:
it's fine to use the JSON big int library here, especially if that makes it possible to remove `replacer` and `reviver`.